platform/linux-generic/include/odp_packet_internal.h

```diff
@@ -248,6 +248,25 @@ static inline uint32_t packet_len(odp_packet_hdr_t *pkt_hdr)
 	return pkt_hdr->frame_len;
 }
 
+static inline uint32_t packet_ref_count(odp_packet_hdr_t *pkt_hdr)
+{
+	/* Breach the atomic type to do a peek at the ref count. This
+	 * is used to bypass atomic operations if ref_count == 1 for
+	 * performance reasons.
+	 */
+	return pkt_hdr->ref_count.v;
+}
+
+static inline void packet_ref_count_set(odp_packet_hdr_t *pkt_hdr, uint32_t n)
+{
+	/* Only used during init when there are no other possible
+	 * references to this pkt, so avoid the "atomic" overhead by
+	 * a controlled breach of the atomic type here. This saves
+	 * over 10% of the pathlength in routines like packet_alloc().
+	 */
+	pkt_hdr->ref_count.v = n;
+}
```

Comment:
TBD later: probably this means that we need inlined atomic operations imlementation.